### DISCUSSION TITLE
Is there a way to run Bevy apps at a lower resolution than native in full-screen mode?

### DISCUSSION THREAD

**Discussion by 66OJ66** [bevyengine/bevy#10738] [Q&A]
Hi,

I'm running my app on a 4k monitor, but I'm trying to work out if I can run it at 1080p instead of 4k (whilst in full-screen mode).

I've tried using the following setup
```rust
fn main() {
    App::new()
        .add_plugins((DefaultPlugins.set(WindowPlugin {
            primary_window: Some(Window {
                resolution: (1920., 1080.).into(),
                mode: WindowMode::Fullscreen,
                ..default()
            }),
            ..default()
        }),))
        .add_systems(Update, display_resolution)
        .run();
}

fn display_resolution(window_query: Query<&Window, With<PrimaryWindow>>) {
    let window = window_query.single();
    let resolution = &window.resolution;

    debug!(
        "Physical Size: ({0}, {1})",
        resolution.physical_width(),
        resolution.physical_height()
    );
    debug!(
        "Logical Size: ({0}, {1})",
        resolution.width(),
        resolution.height()
    );
    debug!("Scale Factor: {}", resolution.scale_factor());
}
```

I'm getting these values logged though:
```
2023-11-25T14:02:02.530350Z DEBUG game: Physical Size: (3840, 2160)
2023-11-25T14:02:02.530363Z DEBUG game: Logical Size: (2194.2856, 1234.2858)
2023-11-25T14:02:02.530368Z DEBUG game: Scale Factor: 1.75
```

So it doesn't look like setting `resolution: (1920., 1080.).into(),` is doing anything?

Am I doing something wrong, or is not possible to use lower-than-native resolutions in full-screen mode currently ?

**Comment by mersenne-twister** [comment]
resolution: (1920., 1080.).into() will just change the physical window size, which if you are using fullscreen, won't do much. you would have to set resolution: WindowResolution::new(3840., 2160.)
                            .with_scale_factor_override(2.),
which will scale it to be twice as big. if you experience blurring, you may need to set ImagePlugin (part of DefaultPlugins)
(note: this works for borderless fullscreen. since fullscreen can be weird, it may be necessary to use borderless to get the desired effect)
